packages/cozy-client/src/models/timeseries.js

```diff
@@ -0,0 +1,74 @@
+import { Q } from '../queries/dsl'
+
+const validateTimeSerieFormat = timeserie => {
+  if (!timeserie.startDate || !timeserie.endDate) {
+    throw new Error(
+      'You must specify a startDate and endDate for the time serie'
+    )
+  }
+  if (!Date.parse(timeserie.startDate) || !Date.parse(timeserie.endDate)) {
+    throw new Error('Invalid date format for the time serie')
+  }
+  if (!timeserie.dataType) {
+    throw new Error('You must specify a dataType for the time serie')
+  }
+  if (!timeserie.serie || !Array.isArray(timeserie.serie)) {
+    throw new Error('You must specify a serie array for the time serie')
+  }
+}
+
+/**
+ * Helper to save a time serie document.
+ *
+ * @param {object} client - The CozyClient instance
+ * @param  {{dataType, serie, startDate, endDate, source, theme}} - The time serie attributes
+ */
+export const saveTimeSerie = async (
+  client,
+  { dataType, serie, startDate, endDate, source, theme }
+) => {
+  validateTimeSerieFormat({ dataType, serie, startDate, endDate, source })
+
+  const doctype = `io.cozy.timeseries.${dataType}`
```

Comment:
I'm not a big fan of not checking the value of `dataType`. Without this check, we can create the doctype we want starting with `io.cozy` and I don't think this is what we want at the end. 

Is it already possible to define a list of dataType or is it too soon?

Reply:
> Without this check, we can create the doctype we want starting with io.cozy
yes, just like the others doctype

> Is it already possible to define a list of dataType
For now, there is just "geojson", but do we want to maintain a predefined list of authorized doctypes in cozy-client ? If we want to restrict the doctypes (and I agree with this part), shoudn't it be enforced by the stack?

Reply:
Update for this: this is something we definitely want in the future, but we merge this PR as it is for now